File: api/java/editor/api-editor/src/main/java/ca/bc/gov/tno/areas/editor/models/TimeTrackingModel.java
package ca.bc.gov.tno.areas.editor.models;

import ca.bc.gov.tno.dal.db.entities.TimeTracking;
import ca.bc.gov.tno.models.AuditColumnModel;

public class TimeTrackingModel extends AuditColumnModel {
  private int contentId;
  private int userId;
  private float effort;
  private String activity;

  public TimeTrackingModel() {
  }

  public TimeTrackingModel(TimeTracking entity) {
    super(entity);

    if (entity != null) {
      this.contentId = entity.getContentId();
      this.userId = entity.getUserId();
      this.effort = entity.getEffort();
      this.activity = entity.getActivity();
    }
  }

  /**
   * @return int the contentId
   */
  public int getContentId() {
    return contentId;
  }

  /**
   * @param contentId the contentId to set
   */
  public void setContentId(int contentId) {
    this.contentId = contentId;
  }

  /**
   * @return int the userId
   */
  public int getUserId() {
    return userId;
  }

  /**
   * @param userId the userId to set
   */
  public void setUserId(int userId) {
    this.userId = userId;
  }

  /**
   * @return float the effort
   */
  public float getEffort() {
    return effort;
  }

  /**
   * @param effort the effort to set
   */
  public void setEffort(float effort) {
    this.effort = effort;
  }

  /**
   * @return String the activity
   */
  public String getActivity() {
    return activity;
  }

  /**
   * @param activity the activity to set
   */
  public void setActivity(String activity) {
    this.activity = activity;
  }

}

File: libs/net/models/Areas/Subscriber/Models/Content/TimeTrackingModel.cs
using TNO.API.Models;

namespace TNO.API.Areas.Subscriber.Models.Content;

/// <summary>
/// TimeTrackingModel class, provides a model that represents an time tracking.
/// </summary>
public class TimeTrackingModel : AuditColumnsModel
{
    #region Properties
    /// <summary>
    /// get/set - Primary key to identify.
    /// </summary>
    public long Id { get; set; }

    /// <summary>
    /// get/set - Foreign key to the content.
    /// </summary>
    public long ContentId { get; set; }

    /// <summary>
    /// get/set - Foreign key to user.
    /// </summary>
    public int UserId { get; set; }

    /// <summary>
    /// get/set - The number of minutes.
    /// </summary>
    public float Effort { get; set; }

    /// <summary>
    /// get/set - The activity performed.
    /// </summary>
    public string Activity { get; set; } = "";
    #endregion

    #region Constructors
    /// <summary>
    /// Creates a new instance of an TimeTrackingModel.
    /// </summary>
    public TimeTrackingModel() { }

    /// <summary>
    /// Creates a new instance of an TimeTrackingModel, initializes with specified parameter.
    /// </summary>
    /// <param name="entity"></param>
    public TimeTrackingModel(Entities.TimeTracking entity) : base(entity)
    {
        this.Id = entity.Id;
        this.ContentId = entity.ContentId;
        this.UserId = entity.UserId;
        this.Effort = entity.Effort;
        this.Activity = entity.Activity;
    }
    #endregion

    #region Methods
    /// <summary>
    /// Creates a new instance of a TimeTracking object.
    /// </summary>
    /// <param name="contentId"></param>
    /// <returns></returns>
    public Entities.TimeTracking ToEntity(long contentId)
    {
        var entity = (Entities.TimeTracking)this;
        entity.ContentId = contentId;
        return entity;
    }

    /// <summary>
    /// Explicit cast to entity.
    /// </summary>
    /// <param name="model"></param>
    public static explicit operator Entities.TimeTracking(TimeTrackingModel model)
    {
        return new Entities.TimeTracking(model.ContentId, model.UserId, model.Effort, model.Activity)
        {
            Id = model.Id,
            Version = model.Version ?? 0,
        };
    }
    #endregion
}
